File: balance.js
const readlineSync = require('readline-sync');
const fs = require('fs');
const utils = require('./utils');

function checkBalance() {
    const accountName = readlineSync.question('Entrez le nom du compte: ');

    const transactions = fs.readFileSync(utils.getTransactionFilePath(), 'utf8');
    const balance = utils.calculateBalance(transactions, accountName);

    if (balance !== undefined) {
        console.log(`Solde actuel du compte "${accountName}": ${balance}`);
    } else {
        console.log(`Le compte "${accountName}" n'existe pas.`);
    }
}

module.exports = {
    checkBalance
};

File: Balance.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.Scanner;

public class Balance {
    public static void checkBalance() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Entrez le nom du compte: ");
        String accountName = scanner.nextLine();

        try {
            String transactions = new String(Files.readAllBytes(Paths.get(Utils.getTransactionFilePath())));
            double balance = Utils.calculateBalance(transactions, accountName);

            if (!Double.isNaN(balance)) {
                System.out.printf("Solde actuel du compte \"%s\": %.2f%n", accountName, balance);
            } else {
                System.out.printf("Le compte \"%s\" n'existe pas.%n", accountName);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
